illumina_uploader/illumina_uploader.py

```diff
@@ -119,65 +122,99 @@ def main():
-                        mailArgs["subject"] = emailInfo["mailsubject"].format(status=status)
-                        mailArgs["body"] = emailInfo["mailbody"].format(folderToUpload=folderToUpload, status=status, timeOfMail=getDateTimeNow(), reason=reason)
-                        if str(emailInfo.get("enabled", None)).lower() == 'true':
-                            sendEmailUsingPlover(emailInfo["emailurl"], mailArgs, logger)
-                        runArgs["runscache"] = runsCache
-                        isSuccessful = False
-                        runArgs["starttime"] = getDateTimeNowIso()
-                        try:
-                            isSuccessful = uploadRunToServer(context, runArgs)
-                        except (socket.error, OSError) as error:
-                            logger.error("Fatal OS / network error: {0}".format(error))
-                            reason = "Fatal OS / network error.. will try again in {0} minutes".format(localInfo["sleeptime"])
-                        status = "FINISHED" if isSuccessful else "FAILED"
-                        logger.info("Marking in DB as {0}".format(status))
-                        dbObject.markFileInDb(folderToUpload, status)
-                        #Create json file
-                        if isSuccessful:
-                            scpUploadCompleteJson(context, runArgs)
-                        #Mail send after done, update subject and body
-                        mailArgs["subject"] = emailInfo["mailsubject"].format(status=status)
-                        mailArgs["body"] = emailInfo["mailbody"].format(folderToUpload=folderToUpload, status=status, timeOfMail=getDateTimeNow(), reason=reason)
-                        if str(emailInfo.get("enabled", None)).lower() == 'true':
-                            sendEmailUsingPlover(emailInfo["emailurl"], mailArgs, logger)
-                #Goto sleep (displayed in minutes)
-                logger.info("Sleeping for {0} minutes".format(localInfo["sleeptime"]))
-                sleeptimeInSeconds = int(localInfo["sleeptime"])*60
-                time.sleep(sleeptimeInSeconds)
-    except FileNotFoundError as error:
-        logger.info(error)
+                    if emailInfo['enabled'] and not shouldStopEmails:
+                        shouldStopEmails = True
+                        subject = emailInfo["mailsubject"].format(status="ERROR")
+                        body = emailInfo["mailbody"].format(folderToUpload="cannot be read", status="ERROR", timeOfMail=getDateTimeNow(), reason=reason)
+                        if ploverEmailInfo["enabled"]:
+                            sendEmailUsingPlover(ploverEmailInfo["emailurl"], { **ploverEmailArgs, 'subject': subject, 'body': body}, logger)
+                        if mcmsEmailInfo["enabled"]:
+                            mcms.send_email(mcmsEmailInfo['emailurl'], mcmsEmailInfo['mailto'], subject, body) 
+                    
+                    time.sleep(sleeptimeInSeconds)
```

Comment:
We won't get to the sleep at the bottom of the main thread so we should sleep here.